src/Nuget.PackageIndex.VisualStudio/CodeFixes/PackageInstaller.cs

```diff
@@ -137,17 +137,20 @@ private void OnPackageInstalled(IVsPackageMetadata metadata)
             // sync at least one of them would do it and add all new packages to the index.
             var indexFactory = new PackageIndexFactory();
             var builder = indexFactory.GetLocalIndexBuilder();
-            builder.BuildAsync(newOnly: true);
+            builder.BuildAsync(newOnly: true, cancellationToken:CancellationToken.None);
         }
 
-        ~PackageInstaller()
+        #region IDisposable
+
+        public void Dispose()
         {
-            Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
-        public void Dispose()
```

Comment:
It is tested already for all scenarios, I just wanted to unsubscribe for events explicitly, this is how it is done In many places , but I agree it looks a bit strange. I just did not see a special place where it is safe to unsubscribe from those events. I will check with Bill if he can suggest something